pkg/controller/controller.go

```diff
@@ -509,6 +514,12 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 		if strings.ToLower(k) == "fstype" {
 			klog.Warningf(deprecationWarning("fstype", prefixedFsTypeKey, ""))
 		}
+		if strings.ToLower(k) == controllerPublishVolumeReadOnly {
```

Comment:
While I am in favor of this PR, I don't like a new parameter in StorageClasses. Do we expect that there will be different classes with different setting for the same driver? What is the use case?

I'd prefer a new cmdline option that turns it on and where CSI driver vendors opts in (or opt out?) the whole driver into this new behavior. Alternatively, a new CSIDriver field would allow cluster admins to opt in / opt out.

Reply:
+1 for cmdline option. That puts the power in the hands of the driver author rather than the admin, and avoids polluting the Kubernetes API.

Opt-out is likely to cause more breakage than opt-in, but opt-out will get us to the desired outcome more quickly.